### PR TITLE
PAS-507 | Python 3.9

### PR DESCRIPTION
**Pull request by jonashendrickx**
### Ticket

- Closes [PAS-507](https://bitwarden.atlassian.net/browse/PAS-507)

### Description

Python 3.8 is reaching EOL, and we need to bump the minimum required version to 3.9.

### Checklist
I did the following to ensure that my changes were tested thoroughly:
- __

I did the following to ensure that my changes do not introduce security vulnerabilities:
- __

[PAS-507]: https://bitwarden.atlassian.net/browse/PAS-507?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: examples/streamlit/pyproject.toml [hunk 1]
```diff
@@ -11,7 +11,7 @@ maintainers = [
 readme = "README.md"
 
 [tool.poetry.dependencies]
-python = "!=3.9.7,>=3.8"
+python = ">=3.9"
 passwordless = "^1"
```
Comment (mzieniukbw) [inline examples/streamlit/pyproject.toml:15]:
```suggestion
passwordless = "^2"
```

File: examples/flask/pyproject.toml [hunk 2]
```diff
@@ -11,7 +11,7 @@ maintainers = [
 readme = "README.md"
 
 [tool.poetry.dependencies]
-python = ">=3.8"
+python = ">=3.9"
 passwordless = "^1"
```
Comment (mzieniukbw) [inline examples/flask/pyproject.toml:15]:
```suggestion
passwordless = "^2"
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -18,10 +18,10 @@ jobs:
...
         run: pipx install poetry
-      - name: Set up Python 3.8
+      - name: Set up Python
         uses: actions/setup-python@f677139bbe7f9c59b41e40162b753c062f5d49a3 # v5.2.0
         with:
-          python-version: '3.8'
+          python-version: '3.9'
           cache: 'poetry'
       - name: Build
         run: poetry install --with dev,test
```
Comment (mzieniukbw) [inline .github/workflows/ci.yml:27]: I am not sure if this is actually using python 3.9. I have only learned recently that the ubuntu-latest GH image already have python installed and poetry will use that one as default and not 3.9.
I would advise adding a step before the `Build`: `poetry env info`.
To fix this, i think we need `poetry env use 3.9`

File: examples/flask/pyproject.toml
Comment (mzieniukbw) [inline examples/flask/pyproject.toml:1]: 💡 how about we add a GH CI workflow (or use existing) for both examples ? At least run `poetry install --with dev` to see if it dependencies are ok.
Note: To use PR's `passwordless` dependency, you might need `poetry install --directory ../..`

Comment (mzieniukbw) [inline examples/flask/pyproject.toml:1]: Update README

File: examples/streamlit/pyproject.toml
Comment (mzieniukbw) [inline examples/streamlit/pyproject.toml:1]: Update README

File: .github/workflows/ci.yml
Comment (mzieniukbw) [inline .github/workflows/ci.yml:1]: `cd.yml` workflow also uses python 3.9. It might also be missing the `poetry env use 3.9` i mentioned here

**Review by mzieniukbw — CHANGES_REQUESTED** [review]
Looks good, but missing GH CD workflow and readme updates

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -37,3 +41,49 @@ jobs:
...
+        with:
+          python-version: '3.9'
+          cache: 'poetry'
+      - name: Setup
+        run: poetry env use 3.9
+      - name: Environment information
+        run: poetry env info
+      - name: Build
+        run: |
+          cd examples/flask;
```
Comment (mzieniukbw) [inline .github/workflows/ci.yml:65]: ⛏️ 
```suggestion
          cd examples/flask
```

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -37,3 +41,49 @@ jobs:
...
+          python-version: '3.9'
+          cache: 'poetry'
+      - name: Setup
+        run: poetry env use 3.9
+      - name: Environment information
+        run: poetry env info
+      - name: Build
+        run: |
+          cd examples/streamlit;
+          poetry install --with dev
```
Comment (mzieniukbw) [inline .github/workflows/ci.yml:89]: new line missing

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -37,3 +41,49 @@ jobs:
...
+        with:
+          python-version: '3.9'
+          cache: 'poetry'
+      - name: Setup
+        run: poetry env use 3.9
+      - name: Environment information
+        run: poetry env info
+      - name: Build
+        run: |
+          cd examples/streamlit;
```
Comment (mzieniukbw) [inline .github/workflows/ci.yml:88]:
```suggestion
          cd examples/streamlit
```

**Review by mzieniukbw — APPROVED** [review]
LGTM

**Review by abergs — APPROVED** [review]

**Pull request merged**